File: src/main/java/mygame/editor/actions/PolygonDrawer.java
package mygame.editor.actions;

import mygame.editor.ui.CustonPane;
import mygame.editor.model.AbstractModel;
import mygame.editor.model.Point;
import mygame.editor.model.PolygonModel;

import java.util.List;

/**
 * Created by oleh on 3/18/17.
 */
public class PolygonDrawer extends Action {

    PolygonModel polygonModel;

    public PolygonDrawer(CustonPane parent, List<AbstractModel> models) {
        super(parent,models);
    }

    @Override
    public void mouseMoved(Point pos) {
        if(polygonModel != null)polygonModel.updateLastPoint(pos);
    }

    @Override
    public void mousePressed(Point pos) {
        if(polygonModel == null){
            polygonModel = new PolygonModel(pos);
            parent.addItem(polygonModel);
        }else{
            polygonModel.addPoint(pos);
        }
    }

    @Override
    public void mouseReleased(Point pos) {

    }

    @Override
    public void finishDrawing() {
        if(polygonModel != null)models.add(polygonModel);
        polygonModel=null;
    }


}

File: src/main/java/mygame/editor/actions/shapes/PolygonDrawer.kt
package mygame.editor.actions.shapes;

import mygame.editor.actions.Action;
import mygame.editor.ui.CustonPane;
import mygame.editor.view.AbstractView;
import mygame.editor.model.Point;
import mygame.editor.view.PolygonModel;


/**
 * Created by oleh on 3/18/17.
 */
class PolygonDrawer (pane:CustonPane,models:List<AbstractView>): Action(pane,models) {

    var polygonModel:PolygonModel? = null


    override fun mouseMoved(pos:Point) {
        polygonModel?.updateLastPoint(pos);
    }

    override fun mousePressed(pos:Point) {
        if(polygonModel == null){
            polygonModel = PolygonModel(pos)
            parent.addItem(polygonModel!!)
        }else{
            polygonModel?.addPoint(pos)
        }
    }

    override fun mouseReleased(pos:Point) {

    }

    override fun finishDrawing() {
        if(polygonModel != null)models.add(polygonModel)
        polygonModel=null
    }


}
